src/lib/sidebar/mergeAdjacentGroups.ts

```
@@ -0,0 +1,53 @@
+ * Recursively merges sidebar groups by combining groups that have only one child
+ * and no direct operation links.
+ */
+export function mergeAdjacentGroups(items: OASidebarItem[]): OASidebarItem[] {
+  if (!Array.isArray(items)) {
+    return items
+  }
+
+  return items.reduce<OASidebarItem[]>((mergedGroups, currentGroup) => {
+    // If group has no items, return it as-is.
+    if (!currentGroup.items?.length) {
+      return [...mergedGroups, currentGroup]
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)